doc/shared/sundomeigest/SUNDomEigEst_API.rst

```diff
@@ -0,0 +1,622 @@
+
+   This *optional* routine should set the number of "warm-up" matrix-vector multiplications,
+   which is executed by :c:func:`SUNDomEig_Estimate` before each estimate.
+
+   **Arguments:**
+
+      * *DEE* -- a SUNDomEigEst object,
+      * *numpreprocess* -- the number of preprocess.
+
+   **Return value:**
+
+      * *numpreprocess* -- the number of preprocessing iterations. 
+
+   **Usage:**
+
+      .. code-block:: c
+
+         retval = SUNDomEigEst_SetNumPreProcess(DEE, numpreprocess);
+
+
+.. c:function:: SUNErrCode SUNDomEigEst_SetTol(SUNDomEigEstimator DEE, sunrealtype tol)
+
+   This *optional* routine sets the estimator's :ref:`relative tolerance <pi_rel_tol>`.
+
+   **Arguments:**
+
+      * *DEE* -- a SUNDomEigEst object,
+      * *tol* -- the requested eigenvalue accuracy.
+
+   **Return value:**
+
+      * *tol* -- the requested eigenvalue accuracy.
+
+   **Usage:**
+
+      .. code-block:: c
+
+         retval = SUNDomEigEst_SetTol(DEE, tol);
+
+
+.. c:function:: SUNErrCode SUNDomEigEst_SetMaxIters(SUNDomEigEstimator DEE, long int max_iters)
+
+   This *optional* routine sets the maximum number of iterations.
+
+   **Arguments:**
+
+      * *DEE* -- a SUNDomEigEst object,
+      * *max_iters* -- the maximum number of iterations.
+
+   **Return value:**
+
+      A :c:type:`SUNErrCode`.
+
+   **Usage:**
+
+      .. code-block:: c
+
+         retval = SUNDomEigEst_SetMaxIters(DEE, max_iters);
+
+
+.. _SUNDomEigEst.GetFn:
+
+SUNDomEigEstimator "get" functions
+----------------------------------
+
+The following functions allow SUNDIALS packages to retrieve results from a
+dominant eigenvalue estimator.  *All routines are optional.*
+
+.. c:function:: SUNErrCode SUNDomEigEst_GetCurRes(SUNDomEigEstimator DEE, sunrealtype* cur_res)
+
+   This *optional* routine should return the final residual from
+   the most-recent "estimator" call.
+
+   **Arguments:**
+
+      * *DEE* -- a SUNDomEigEst object.
+      * *cur_res* -- the current residual
+
+   **Return value:**
+
+      A :c:type:`SUNErrCode`.
+
+   **Usage:**
+
+      .. code-block:: c
+
+         sunrealtype cur_res;
+         retval = SUNDomEigEst_GetCurRes(DEE, &cur_res);
+
+
+.. c:function:: SUNErrCode SUNDomEigEst_GetCurNumIters(SUNDomEigEstimator DEE, long int* curniter)
+
+   This *optional* routine should return the number of estimator
+   iterations performed in the most-recent "estimator" call.
+
+   **Arguments:**
+
+      * *DEE* -- a SUNDomEigEst object,
+      * *curniter* -- the current number of iterations.
+
+   **Return value:**
+
+      A :c:type:`SUNErrCode`.
+
+   **Usage:**
+
+      .. code-block:: c
+
+         long int curniter;
+         retval = SUNDomEigEst_GetCurNumIters(DEE, &curniter);
+
+
+.. c:function:: SUNErrCode SUNDomEigEst_GetMaxNumIters(SUNDomEigEstimator DEE, long int* max_niter)
+
+   This *optional* routine should return the maximum number of iterations
+   performed in all "estimator" calls so far.
```

Comment:
Is this the maximum number of iterations in any single "Estimate" call so far, or the total number of iterations used in all "Estimate" calls so far?  If its the former, then I recommend:
```suggestion
.. c:function:: SUNErrCode SUNDomEigEst_GetMaxNumIters(SUNDomEigEstimator DEE, long int* max_niter)

   This *optional* routine should return the maximum number of iterations
   performed in any previous "estimator" call so far.
```